apps/confidential/confidential.go

```diff
@@ -531,15 +583,17 @@ func WithSilentAccount(account Account) interface {
 
 // AcquireTokenSilent acquires a token from either the cache or using a refresh token.
 //
-// Options:
-//   - [WithSilentAccount]
-//   - [WithTenantID]
+// Options: [WithClaims], [WithSilentAccount], [WithTenantID]
 func (cca Client) AcquireTokenSilent(ctx context.Context, scopes []string, opts ...AcquireSilentOption) (AuthResult, error) {
 	o := AcquireTokenSilentOptions{}
 	if err := options.ApplyOptions(&o, opts); err != nil {
 		return AuthResult{}, err
 	}
 
+	if o.claims != "" {
```

Comment:
You raise a good point. It will one day be possible for this method to succeed when given claims; we can add the option then. The value of taking the option today is in maintaining the calling pattern used in all other cases (see e.g. [azidentity](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/client_secret_credential.go#L51-L62)):
1. call `AcquireTokenSilent` (check the cache)
2. if that returned an error, call `AcquireTokenBy*` (request a new token)

Applications expect errors from `AcquireTokenSilent` because it commonly returns them, which is okay because that's just a cache miss. Having an option that guarantees it will return an error therefore doesn't make users write more, different, or tricky code. On the other hand, if this method doesn't take claims, the probably miniscule population of users who use this feature will have to write conditional code around it, then rewrite that code when this method does take claims. But sure, it's weird to have an anti-useful option.

Okay, how do you feel about it now? I lean toward keeping the option but might flop to the other side of the fence if you push hard enough 😆

Reply:
I think I get what you mean - it's definitely not a straightforward case and it sounds like it's critical to consider that it's running as part of an overall pattern of calls, not just on its own (if I'm following your logic).

Resolving for now, explanation makes sense to me.